hack/update-ubuntu-image.sh

```diff
@@ -0,0 +1,324 @@
+				[[ -v ubuntu_image_url_latest_cache[${latest_cache_key}] ]] && echo "${ubuntu_image_url_latest_cache[${latest_cache_key}]}" ||
+					ubuntu_image_url_latest "${version}" "${flavor}" "${arch}" "${path_suffix}"
+			)
+			ubuntu_image_url_latest_cache[${latest_cache_key}]="${location_digest}"
+			read -r location digest <<<"${location_digest}"
+			if [[ -z ${location} ]]; then
+				echo "Failed to get the latest image location for ${location_basename}" >&2
+				continue
+			elif [[ ${location} == "${location_before}" ]]; then
+				continue
+			fi
+			image_entry="{\"location\": \"${location}\", \"arch\": \"${limayaml_arch}\", \"digest\": \"${digest}\"}"
+			echo -e "${location}\n${digest}"
+			if [[ ${kernel_location} != "null" ]]; then
+				kernel_info=$(ubuntu_kernel_info_for_image_url "${location}")
+				IFS=$'\t' read -r kernel_location kernel_digest initrd_location initrd_digest <<<"${kernel_info}"
+				if [[ -n ${kernel_location} ]]; then
+					image_entry=$(jq ". + {kernel: {location: \"${kernel_location}\", digest: \"${kernel_digest}\"}}" <<<"${image_entry}")
+					[[ ${kernel_cmdline} != "null" ]] && image_entry=$(jq ".kernel.cmdline = \"${kernel_cmdline}\"" <<<"${image_entry}")
+					echo -e "${kernel_location}\n${kernel_digest}"
+				fi
+				if [[ -n ${initrd_location} ]]; then
+					image_entry=$(jq ". + {initrd: {location: \"${initrd_location}\", digest: \"${initrd_digest}\"}}" <<<"${image_entry}")
+					echo -e "${initrd_location}\n${initrd_digest}"
+				fi
+			fi
+		else
+			release_cache_key=${version}-${flavor}-${arch}-${path_suffix}
+			location=$(
+				# shellcheck disable=SC2015
+				[[ -v ubuntu_image_url_release_cache[${release_cache_key}] ]] && echo "${ubuntu_image_url_release_cache[${release_cache_key}]}" ||
+					ubuntu_image_url_release "${version}" "${flavor}" "${arch}" "${path_suffix}"
+			)
+			ubuntu_image_url_release_cache[${release_cache_key}]="${location}"
+			if [[ -z ${location} ]]; then
+				echo "Failed to get the release image location for ${location_basename}" >&2
+				continue
+			elif [[ ${location} == "${location_before}" ]]; then
+				continue
+			fi
+			image_entry="{\"location\": \"${location}\", \"arch\": \"${limayaml_arch}\"}"
+			echo "${location}"
+			if [[ ${kernel_location} != "null" ]]; then
+				kernel_info=$(ubuntu_kernel_info_for_image_url "${location}")
+				IFS=$'\t' read -r kernel_location kernel_digest initrd_location initrd_digest <<<"${kernel_info}"
+				if [[ -n ${kernel_location} ]]; then
+					image_entry=$(jq ". + {kernel: {location: \"${kernel_location}\"}}" <<<"${image_entry}")
+					[[ ${kernel_cmdline} != "null" ]] && image_entry=$(jq ".kernel.cmdline = \"${kernel_cmdline}\"" <<<"${image_entry}")
+					echo "${kernel_location}"
+				fi
+				if [[ -n ${initrd_location} ]]; then
+					image_entry=$(jq ". + {initrd: {location: \"${initrd_location}\"}}" <<<"${image_entry}")
+					echo "${initrd_location}"
+				fi
+			fi
+		fi
+		limactl edit --log-level error --set "
```

Comment:
Can we just use yq?

Reply:
Why? Wouldn't it still essentially be the same command?

By using `limactl` we have a test case that `limactl edit` can edit template files directly, as well as instances.

Reply:
It's no exaggeration to say that #2593 was created for this script.